doc/system_requirements.md

```diff
@@ -0,0 +1,124 @@
+# System Requirement Specification Advanced Analytics Framework
+
+## Introduction
+
+This library is an event-driven framework that allows users to build complex 
+data processing algorithms in Exasol.
+
+
+## About This Document
+
+### Goal
+
+The goal of this library is to provide a general framework to implement complex 
+data analysis algorithms with Exasol. This framework provides certain features 
+to users in which they are able to run their implementations.
+
+## Stakeholders
+
+Software developers use this framework to develop complex data processing 
+algorithms such as machine learning methods.
+
+## Features
+
+Features are the highest level requirements in this document that describe the 
+main functionality of Advanced Analytics Framework.
+
```

Comment:
The algorithm is implemented by the user, so the interface needs to be simple for everyone to implement algorithms